experimenter/experimenter/experiments/tests/api/v6/test_views.py

```diff
@@ -179,3 +182,89 @@ def test_list_view_serializes_live_first_run_experiments(self):
         json_slugs = [d["id"] for d in json_data]
         expected_slugs = [e.slug for e in experiments]
         self.assertEqual(json_slugs, expected_slugs)
+
+
+class TestNimbusFmlDiagnosticsApi(TestCase):
+    maxDiff = None
+
+    # todo: replace the hardcoded return values with something parsable by the endpoint
+    fml_diagnostics = [
+        {
+            "line": 0,
+            "col": 0,
+            "message": "Incorrect value",
+            "highlight": "enabled",
+        },
+        {
+            "line": 1,
+            "col": 2,
+            "message": "Type not allowed",
+            "highlight": "record",
+        },
+    ]
+
+    @mock.patch("experimenter.experiments.api.v6.views.NimbusFmlDiagnosticsApi.post")
+    def test_get_fml_diagnostics(self, mock_fetch_fml_diagnostics):
+        test_blob = "hello world"
+        fml_response = Response()
+        fml_response.content = self.fml_diagnostics
+        fml_response.status_code = status.HTTP_201_CREATED
+
+        mock_fetch_fml_diagnostics.return_value = fml_response
+
+        response = self.client.post(
+            reverse("nimbus-fml-diagnostics-api"),
+            kwargs={
+                "application": "firefox_ios",
+                "channel": "release",
+                "blob": test_blob,
+            },
+        )
+        self.assertEqual(response.status_code, 201)
+        result = response.content.decode("utf-8")
+        self.assertIn(self.fml_diagnostics[0].get("message"), result)
+        self.assertIn(self.fml_diagnostics[1].get("message"), result)
+
+    @mock.patch("experimenter.experiments.api.v6.views.NimbusFmlDiagnosticsApi.post")
+    def test_fml_diagnostics_endpoint_send_correct_args(self, mock_fetch_fml_diagnostics):
+        test_blob = json.dumps(
+            {
+                "features": {
+                    "upgrade-message": {
+                        "description": "A message displayed when the user upgrades",
+                        "variables": {
+                            "hero-image": {
+                                "description": "A pre-bundled image",
+                                "type": "Image",
+                                "default": "ic_fox",
+                            },
+                            "message-content": {
+                                "description": "Text content to show the user",
+                                "default": "msg_thankyou",
+                            },
+                        },
+                    }
+                }
+            }
```

Comment:
This look _like_ a sample FML in JSON. Is this necessarily complicated, or can you use a simpler string for this test case?

Reply:
Absolutely can 👍